### ISSUE TITLE
path in assets GET could be a regex or substring filtering

### ISSUE THREAD

**Issue by satra** [dandi/dandi-api#230]
it would be nice to be able to filter quickly on items in a dataset by any substring. technically on the db side this should be a simple query on the path column. 

instead of 
```
https://api.dandiarchive.org/api/dandisets/000003/versions/draft/assets/?path=sub-YutaMouse20
```
we should also be able to do:
```
https://api.dandiarchive.org/api/dandisets/000003/versions/draft/assets/?path=YutaMouse20
```
and
```
https://api.dandiarchive.org/api/dandisets/000003/versions/draft/assets/?path=behavior
```

one possibility is to change `path` to `path-contains`

**Comment by yarikoptic** [comment]
I don't think it should be that implicit -- if you are aiming for a regex, then let's make it into an extra flag (`regex=False|True`) or even extra option (`regex=String`) which could be used along with the `path` where `path` would then restrict the top level and the candidates are filtered using `regex`.

**Comment by waxlamp** [comment]
@danlamanna, could we accomplish this mode of search via a trigram index, falling back to some kind of full search for search terms shorter than 3 characters?
